iris_ugrid/tests/synthetic_data_generator/__init__.py

```diff
@@ -0,0 +1,115 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Provides synthetic data generation capability in order to test ugrid loading and regridding.
+"""
+from pathlib import Path
+from string import Template
+import subprocess
+
+import netCDF4
+import numpy as np
+
+
+def create_file(
+    temp_file_dir,
+    dataset_name,
+    dataset_type,
+    n_faces=866,
```

Comment:
I'm worried that at least these last 3 control args are general enough to cope with future new `dataset_type`s.
Can we make these `kwargs` ??
Then you need to pass a template, and substitution vars appropriate to the template.

I think we can accept that they all potentially use "dataset_name", but the other template variables are determined by the specific template (== dataset type)

Reply:
Actually, an edge-based XIOS-like dataset is a good example of this : As it is, that case would require us to expand this routine with an `n_edges` arg (and it will have no use for `n_faces`)

Reply:
See b62b005f6ba69f98a48a0f9d1ec35028e39c3aba for the eventual solution.